nautobot/extras/views.py

```diff
@@ -355,6 +363,216 @@ class ConfigContextSchemaBulkDeleteView(generic.BulkDeleteView):
     filterset = filters.ConfigContextSchemaFilterSet
 
 
+#
+# Contacts
+#
+
+
+class ContactUIViewSet(NautobotUIViewSet):
+    bulk_update_form_class = forms.ContactBulkEditForm
+    filterset_class = filters.ContactFilterSet
+    filterset_form_class = forms.ContactFilterForm
+    form_class = forms.ContactForm
+    queryset = Contact.objects.all()
+    serializer_class = serializers.ContactSerializer
+    table_class = tables.ContactTable
+    is_contact_model = False
+
+    def get_extra_context(self, request, instance):
+        context = super().get_extra_context(request, instance)
+        if self.action == "retrieve":
+            teams = instance.teams.restrict(request.user, "view")
+            teams_table = tables.TeamTable(teams, orderable=False)
+            teams_table.columns.hide("actions")
+            paginate = {"paginator_class": EnhancedPaginator, "per_page": get_paginate_count(request)}
+            RequestConfig(request, paginate).configure(teams_table)
+            context["teams_table"] = teams_table
+
+            # TODO: need some consistent ordering of contact_associations
+            associations = instance.contact_associations.restrict(request.user, "view")
+            associations_table = tables.ContactAssociationTable(associations, orderable=False)
+            RequestConfig(request, paginate).configure(associations_table)
+            context["contact_associations_table"] = associations_table
+        return context
+
+
+class ContactAssociationUIViewSet(ObjectBulkDestroyViewMixin, ObjectDestroyViewMixin, ObjectEditViewMixin):
+    form_class = forms.ContactAssociationForm
+    filterset_class = filters.ContactAssociationFilterSet
+    queryset = ContactAssociation.objects.all()
+    serializer_class = serializers.ContactAssociationSerializer
+    table_class = AssociatedContactsTable
+    non_filter_params = ("export", "page", "per_page", "sort")
+
+
+class ObjectNewContactView(generic.ObjectEditView):
+    queryset = Contact.objects.all()
+    model_form = forms.ObjectNewContactForm
+    template_name = "extras/object_new_contact.html"
+
+    def successful_post(self, request, obj, created, new_logger):
+        """Callback after the form is successfully saved but before redirecting the user."""
+        verb = "Created" if created else "Modified"
+        msg = f"{verb} {self.queryset.model._meta.verbose_name} and Created Contact Association"
```

Comment:
This feels to me like maybe it's exposing too much of the system "guts" to the end user. They don't really care that Contact Association is a separate part of the data model? Same comment below.